Bug: http://bugs.mysql.com/bug.php?id=62769
Bug-Ubuntu: http://pad.lv/700982
Description: make sure va_list usage is portable to fix FTBFS on ARM
Reviewed-by: Nicholas Bamber <nicholas@periapt.co.uk>
Last-Update: 2012-10-24

Index: mysql-5.5.29/sql-common/client_plugin.c
===================================================================
--- mysql-5.5.29.orig/sql-common/client_plugin.c	2013-01-02 15:36:15.258361966 -0800
+++ mysql-5.5.29/sql-common/client_plugin.c	2013-01-02 15:38:12.035514668 -0800
@@ -233,11 +233,13 @@
 {
   MYSQL mysql;
   struct st_mysql_client_plugin **builtin;
+  va_list unused;
 
   if (initialized)
     return 0;
 
   bzero(&mysql, sizeof(mysql)); /* dummy mysql for set_mysql_extended_error */
+  bzero(&unused, sizeof(unused)); /* suppress uninitialized-value warnings */
 
   pthread_mutex_init(&LOCK_load_client_plugin, MY_MUTEX_INIT_SLOW);
   init_alloc_root(&mem_root, 128, 128);
@@ -249,7 +251,7 @@
   pthread_mutex_lock(&LOCK_load_client_plugin);
 
   for (builtin= mysql_client_builtins; *builtin; builtin++)
-    add_plugin(&mysql, *builtin, 0, 0, 0);
+    add_plugin(&mysql, *builtin, 0, 0, unused);
 
   pthread_mutex_unlock(&LOCK_load_client_plugin);
 
@@ -293,9 +295,13 @@
 mysql_client_register_plugin(MYSQL *mysql,
                              struct st_mysql_client_plugin *plugin)
 {
+  va_list unused;
+
   if (is_not_initialized(mysql, plugin->name))
     return NULL;
 
+  bzero(&unused, sizeof(unused)); /* suppress uninitialized-value warnings */
+
   pthread_mutex_lock(&LOCK_load_client_plugin);
 
   /* make sure the plugin wasn't loaded meanwhile */
@@ -307,7 +313,7 @@
     plugin= NULL;
   }
   else
-    plugin= add_plugin(mysql, plugin, 0, 0, 0);
+    plugin= add_plugin(mysql, plugin, 0, 0, unused);
 
   pthread_mutex_unlock(&LOCK_load_client_plugin);
   return plugin;
